pkg/operator/operators/klusterlet/controllers/klusterletcontroller/klusterlet_controller_test.go

```diff
@@ -463,6 +463,11 @@ func ensureObject(t *testing.T, object runtime.Object, klusterlet *operatorapiv1
 		return
 	}
 
+	if !mapCompare(helpers.MergeLabels(helpers.KlusterletAgentDefaultLabels(), klusterlet.GetLabels()), access.GetLabels()) {
```

Comment:
I think you can just use deepEqual?

Reply:
the synced klusterlet labels is a part of the exiting labels, for example the deployments have label `app=klusterlet-registration-agent`, namespace has the label `workload.openshift.io/allowed=management`